src/s3ql/backends/b2/b2_backend.py

```diff
@@ -0,0 +1,807 @@
+                else:
+                    keys_remaining = False
+                    break
+
+        return versions
+
+    @retry
+    def _list_file_versions_page(self, next_filename=None, next_file_id=None):
+        request_dict = {
+            # maximum is 10000, but will get billed in steps of 1000
+            'maxFileCount': 1000,
+            'bucketId': self._get_bucket_id()
+        }
+
+        if next_filename is not None:
+            request_dict['startFileName'] = next_filename
+
+        if next_file_id is not None:
+            request_dict['startFileId'] = next_file_id
+
+        response = self._do_api_call('b2_list_file_versions', request_dict)
+        file_versions_list = [ { 'fileName': file_version['fileName'], 'fileId': file_version['fileId'] } for file_version in response['files'] ]
+
+        return response['nextFileName'], response['nextFileId'], file_versions_list
+
+    @copy_ancestor_docstring
+    def list(self, prefix=''):
+        next_filename = self._get_key_with_prefix(prefix)
+        keys_remaining = True
+
+        while keys_remaining and next_filename is not None:
+            next_filename, filelist = self._list_file_names_page(next_filename)
+
+            for file_ in filelist:
+                if file_.startswith(self.prefix + prefix):
+                    # remove prefix before return
+                    r = file_[len(self.prefix):]
+                    decoded_r = self._b2_url_decode(r, decode_plus=False)
+                    yield decoded_r
+                else:
+                    keys_remaining = False
+                    break
+
+    @retry
+    def _list_file_names_page(self, next_filename=None):
+        request_dict = {
+            # maximum is 10000, but will get billed in steps of 1000
+            'maxFileCount': 1000,
+            'bucketId': self._get_bucket_id()
+        }
+
+        if next_filename is not None:
+            request_dict['startFileName'] = next_filename
+
+        response = self._do_api_call('b2_list_file_names', request_dict)
+        filelist = [ file_['fileName'] for file_ in response['files'] ]
+
+        return response['nextFileName'], filelist
+
+    def _get_file_id_and_name(self, key):
+        head_request_response = self._do_download_request('HEAD', key)
+        file_id = self._b2_url_decode(head_request_response.headers['X-Bz-File-Id'])
+        file_name = self._b2_url_decode(head_request_response.headers['X-Bz-File-Name'])
+        file_name = file_name.replace('\\', '=5C')
+        return file_id, file_name
+
+    @retry
+    @copy_ancestor_docstring
+    def copy(self, src, dest, metadata=None):
+        log.debug('started with %s, %s', src, dest)
+
+        source_file_id, source_file_name = self._get_file_id_and_name(src)
+
+        request_dict = {
+            'sourceFileId': source_file_id,
+            'fileName': self.prefix + dest.replace('\\', '=5C') # self._get_key_with_prefix(dest)
```

Comment:
Because `_get_key_with_prefix` uses `_b2_url_encode` which in turn calls `urllib.parse.quote`, but in this case, the parameters are passed as JSON to B2, where only the backslashes have to be converted but not the rest.